backend/spec/controllers/spree/admin/orders/customer_details_controller_spec.rb

```diff
@@ -20,23 +20,175 @@
+              expect(order).to receive(:refresh_shipment_rates)
+              .with(Spree::ShippingMethod::DISPLAY_ON_FRONT_AND_BACK_END)
+            end
+            it { expect(controller).to receive(:load_order).and_call_original }
+            it { expect(controller).to receive(:guest_checkout?).twice.and_call_original }
+            it { expect(controller).to_not receive(:load_user).and_call_original }
+            after { send_request(attributes) }
+          end
+        end
+
+        context 'having invalid parameters' do
+          before do
+            allow(order).to receive_messages(update_attributes: false)
+          end
+
+          context 'having failure response' do
+            before { send_request(attributes) }
+            it { expect(response).to render_template(:edit) }
+          end
+
+          context 'with correct method flow' do
+            it { expect(order).to receive(:update_attributes).with(attributes[:order]) }
+            it { expect(controller).to receive(:load_order).and_call_original }
+            it { expect(controller).to receive(:guest_checkout?).and_call_original }
+            it { expect(controller).to_not receive(:load_user).and_call_original }
+            after { send_request(attributes) }
+          end
+        end
+      end
+
+      context 'without using guest checkout' do
+        let (:user) { mock_model(Spree.user_class) }
+        let (:_attributes) { attributes.merge(guest_checkout: 'false', user_id: user.id) }
+
+        context 'having valid parameters' do
+          before do
+            allow(Spree.user_class).to receive(:find_by).and_return(user)
+            allow(order).to receive_messages(update_attributes: true)
+            allow(order).to receive_messages(next: false)
+            allow(order).to receive_messages(complete?: true)
+            allow(order).to receive_messages(refresh_shipment_rates: true)
+            allow(order).to receive_messages(associate_user!: true)
+            allow(controller).to receive(:guest_checkout?).and_return(false)
+            allow(order).to receive(:associate_user!)
+          end
+
+          context 'having successful response' do
+            before { send_request(_attributes) }
+            it { expect(response).to have_http_status(302) }
+            it { expect(response).to redirect_to(edit_admin_order_url(order)) }
+          end
+
+          context 'with correct method flow' do
+            it { expect(order).to receive(:update_attributes).with(_attributes[:order]) }
+            it { expect(order).to receive(:associate_user!).with(user, order.email.blank?) }
+            it { expect(order).to_not receive(:next) }
+            it { expect(order).to receive(:complete?) }
+            it 'does refresh the shipment rates with all shipping methods' do
+              expect(order).to receive(:refresh_shipment_rates)
+              .with(Spree::ShippingMethod::DISPLAY_ON_FRONT_AND_BACK_END)
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Use 2 (not 0) spaces for indenting an expression spanning multiple lines.